packages/database/src/core/util/libs/parser.ts

```diff
@@ -174,17 +174,18 @@ export const parseDatabaseURL = function(
       colonInd = dataURL.length;
     }
 
-    const parts = host.split('.');
-    if (parts.length === 3) {
+    let dotInd = host.indexOf('.');
+    if (dotInd === -1) {
+      dotInd = colonInd;
+    }
+    const hostFirstPart = host.substring(0, dotInd);
+    if (hostFirstPart.toLowerCase() === 'localhost') {
+      domain = 'localhost';
+    } else {
+      domain = host.substring(dotInd+1);
       // Normalize namespaces to lowercase to share storage / connection.
-      domain = parts[1];
-      subdomain = parts[0].toLowerCase();
-      // We interpret the subdomain of a 3 component URL as the namespace name.
+      subdomain = hostFirstPart.toLowerCase();
       namespace = subdomain;
-    } else if (parts.length === 2) {
-      domain = parts[0];
```

Comment:
This case would be changed if we generalized it.
`foo.bar` ==> `namespace = foo, domain = bar`